OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1607,6 +1607,43 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				// Detonates replaces DemoTruck and changes MadTank
+				if (engineVersion < 20150718)
+				{
+					if (node.Key.StartsWith("DemoTruck"))
+					{
+						node.Key = "Detonates";
+						var voice = node.Value.Nodes.FirstOrDefault(w => w.Key == "Voice");
+
+						if (voice != null)
+							voice.Key = "OrderVoice";
+					}
+
+					if (node.Key.StartsWith("MadTank"))
+					{
+						node.Value.Nodes.RemoveAll(n =>
+							n.Key == "DetonationDelay" ||
+							n.Key == "DetonationSound" ||
+							n.Key == "DetonationWeapon" ||
+							n.Key == "ChargeDelay" ||
+							n.Key == "ChargeSound" ||
+							n.Key == "Voice");
+					}
+
+					if (node.Value.Nodes.Any(n => n.Key == "MadTank"))
+					{
+						node.Value.Nodes.Add(new MiniYamlNode("Detonates", "", new List<MiniYamlNode>
+						{
+							new MiniYamlNode("ChargeDelay", "96"),
+							new MiniYamlNode("ChargeSound", "madchrg2.aud"),
+							new MiniYamlNode("DetonationDelay", "42"),
+							new MiniYamlNode("DetonationSound", "MadExplo.aud"),
```

Comment:
Likewise here: you must be consistent with filename case on disk.  We don't all run windows!